### PR TITLE
iam-policy-json-to-terraform 1.3.0 (new formula)

### PR DESCRIPTION
**Pull request by nitrocode**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/iam-policy-json-to-terraform.rb
Comment (SMillerDev) [inline Formula/iam-policy-json-to-terraform.rb]: We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

File: Formula/iam-policy-json-to-terraform.rb
Comment (SMillerDev) [inline Formula/iam-policy-json-to-terraform.rb]: Does this download any dependencies?

File: Formula/iam-policy-json-to-terraform.rb
Comment (SMillerDev) [inline Formula/iam-policy-json-to-terraform.rb]: What's special about the macOS make?

File: Formula/iam-policy-json-to-terraform.rb
Comment (nitrocode) [reply Formula/iam-policy-json-to-terraform.rb]: Thanks for reviewing. I added a more detailed input and output check to verify cli functionality.

File: Formula/iam-policy-json-to-terraform.rb
Comment (nitrocode) [reply Formula/iam-policy-json-to-terraform.rb]: Yes, this is extraneous, good point. I've removed this

File: Formula/iam-policy-json-to-terraform.rb
Comment (nitrocode) [reply Formula/iam-policy-json-to-terraform.rb]: This `make` command runs `GOOS=darwin go build -o $@ *.go` but when I replace this line with

```
system "go", "build", "-o", "iam-policy-json-to-terraform", "*.go"
```

it returns

```
go
build
-o
iam-policy-json-to-terraform
*.go

can't load package: package *.go: cannot find package "*.go" in any of:
	/usr/local/Cellar/go/1.13.7/libexec/src/*.go (from $GOROOT)
	/private/tmp/iam-policy-json-to-terraform-20200209-22723-pqecfk/iam-policy-json-to-terraform-1.3.0/src/*.go (from $GOPATH)

READ THIS: https://docs.brew.sh/Troubleshooting
```

Could you recommend what to do?

File: Formula/iam-policy-json-to-terraform.rb
Comment (SMillerDev) [inline Formula/iam-policy-json-to-terraform.rb]: Please just write this as a simple json string.

File: Formula/iam-policy-json-to-terraform.rb
Comment (SMillerDev) [inline Formula/iam-policy-json-to-terraform.rb]: This doesn't add anything and should be removed

File: Formula/iam-policy-json-to-terraform.rb
Comment (SMillerDev) [inline Formula/iam-policy-json-to-terraform.rb]: Is that single word al that's guaranteed to be in the output?

**Review by SMillerDev — COMMENTED** [review]
Some more comnents

File: Formula/iam-policy-json-to-terraform.rb
Comment (nitrocode) [reply Formula/iam-policy-json-to-terraform.rb]: I tried doing an `assert_match` on multiple lines of output and it kept raising an error. Would you know the best way to verify multiline output?

```
✗ echo '{"Statement":[{"Effect":"Allow","Action":["ec2:Describe*"],"Resource":"*"}]}' | iam-policy-json-to-terraform
data "aws_iam_policy_document" "policy" {
  statement {
    sid       = ""
    effect    = "Allow"
    resources = ["*"]
    actions   = ["ec2:Describe*"]
  }
}
```

File: Formula/iam-policy-json-to-terraform.rb
Comment (nitrocode) [reply Formula/iam-policy-json-to-terraform.rb]: I'm not super familiar of ruby. Would you suggest an example formula with the json string style that's wanted here? Is it just like this?

```
    test_input = "{ \"Statement\": \"blah\" }"
```

File: Formula/iam-policy-json-to-terraform.rb
Comment (nitrocode) [reply Formula/iam-policy-json-to-terraform.rb]: Thanks. I removed it.

File: Formula/iam-policy-json-to-terraform.rb
Comment (nitrocode) [reply Formula/iam-policy-json-to-terraform.rb]: I've replaced it with a simple single quoted string so I do not have to do the quoted escapes.

File: Formula/iam-policy-json-to-terraform.rb
Comment (nitrocode) [reply Formula/iam-policy-json-to-terraform.rb]: I've changed it to check for `ec2:Describe*` instead which is a better check for the efficacy of the tool's output.

File: Formula/iam-policy-json-to-terraform.rb
Comment (nitrocode) [reply Formula/iam-policy-json-to-terraform.rb]: I dropped the dependency checks since brew takes care of that

```
       system "make", "vendor", "tools" 
```

and kept the original make command that I'm having trouble replicating within homebrew itself.

```
      system "make", "iam-policy-json-to-terraform_darwin"
```

Would this suffice @SMillerDev ?

**Comment by nitrocode** [comment]
All comments resolved. Is there anything else I need to do before this can be approved and merged?

**Comment by SMillerDev** [comment]
Yes, be patient. Homebrew maintainers are volunteers. You and 100+ other pull requests made today will all get attention when someone gets to you.

**Comment by nitrocode** [comment]
@SMillerDev My mistake; I didn't mean to sound impatient. I just wanted to make sure I have all the ducks in a row to get this accepted. If not, I can make the necessary changes.

Regarding criteria for popularity, [iam-policy-json-to-terraform](https://github.com/flosell/iam-policy-json-to-terraform) has 2 watchers, 102 stars, and 11 forks. Is this popular enough to enter homebrew?

**Review by chenrui333 — APPROVED** [review]
lgtm

Maybe you guys should also think to move to go-modules :)

**Comment by nitrocode** [comment]
@chenrui333 thank you! I'm still new to golang but I'll look up go-modules. It's possible that this app is already using go-modules as it's a single golang file, no? @flosell care to weigh in on that?

https://github.com/flosell/iam-policy-json-to-terraform

**Comment by flosell** [comment]
hey @nitrocode, as I understand it, go-modules would be a replacement of `dep` as a dependency management tool that has a number of benefits (being the de-facto default for new projects among others) but also a slightly different way of doing things. 

I looked into it a few months ago and in the end decided it wasn't worth investing effort into the change at this time. I'm open to be convinced otherwise (or simply exchange thoughts in more detail), feel free to open an issue on the project.

**Comment by nitrocode** [comment]
Thanks for the comment @flosell.

The traditional way to drop the use of the custom `make` for golang formula, from what I've seen, is by using the `go_dependency` function but according to the audit, that has been deprecated in favor of proper vendoring. We can create a ticket in the backlog to do proper vendoring of go dependencies and that should allow dropping the need to run the `make` command.

For now, lgtm. 😄

**Pull request closed without merging**
